### ISSUE TITLE
[URGENT!!!!] iOS Integration with existing apps is not correct ! 

### ISSUE THREAD

**Issue by edgarclerigo** [facebook/react-native-website#1194]
Here:

https://facebook.github.io/react-native/docs/integration-with-existing-apps

Following those steps will lead you to a few errors, which might lead developers to lost a lot of time trying to find solutions for this ! Please update this page ! 

For example, the section

`Configuring CocoaPods dependencies`

On the Podfile, this part is incorrect 

```
pod 'React', :path => '../node_modules/react-native', :subspecs => [
    'Core',
    'CxxBridge', # Include this for RN >= 0.47
    'DevSupport', # Include this to enable In-App Devmenu if RN >= 0.43
    'RCTText',
    'RCTNetwork',
    'RCTWebSocket', # needed for debugging
    # Add any other subspecs you want to use in your project
  ]
```

If i create a new react-project using `react-native init MyAwesomeProject` , if I go to the Podfile automatically generated, the Podfile is the correct one, with the correct paths !

Using `use_frameworks!` will lead you to an error on compile process, already reported here, 

https://github.com/facebook/react-native/issues/26124

After using the right Podfile and by removing the `use_frameworks!` , and call `pod install`, when I tried to `import React` it fails, and the only way I found to make this working, was by creating a `Bridging-Header-h` file ....

**Comment by Metroidaron** [comment]
## This is still an issue! 

### tldr; Yes the above was an issue; the guide for integrating React-Native into an existing app has been a very rough experience and taken a LOT of time; Can someone who has done this integration before verify there are no other big breakages inside the integration guide like this one?  Luckily I've been using react-native init as a reference for troubleshooting; but pods is still very frustrating... 

Some of this probably doesn't belong here, but I'm just trying to give a Full Picture in case someone more experienced with integrating can point at a single spot and say:

> "Right here is where Everything went wrong!" 

Long story short (not so much now); I'm the react-native guy at my work; and we also have a native team.  I've been working for the past week trying to upgrade their react-native component from a really old version to up to date (The package they had embedded updated their stuff so we're having to update ours, they do no work in react-native, its only implemented to include a library/third party interface done by an outside company).  

I've never integrated react into an already existing project (which didn't use pods; they manually linked everything in the past, **Is that even still an option these days???**)  Any ways, getting them up and running has been a nightmare because we finally broke down and started trying to use pods and create a simple Vanilla no content react-native integration.  I figured out the above pretty quickly (It only appears on the swift version of the guide) and would love for someone else to verify that nothing else is broken in this guide... 

Our issues might arise from a different project structure than what react-native init creates (Which is what I do, so i'll list the big differences below, if any are a known deal breaker for getting react working; please let me know), but we have spent the last day going down the rabbit hole of trying to update links and environment variables that pods got wrong or just didn't do (For example, adding PODS_ROOT and other variables that weren't defined...) and our latest blocker we haven't found our way around is `No such module 'React'` when importing React into the swift file.  
- Already existing workspace exists one dir up from project file; so we created a podfile next to the project file, not the workspace
- Team doesn't use pods, but rather Carthage; we broke down and tried to get pods up and running but was a broken experience
- The native guys went in and had to make a lot of small changes getting rid of errors one by one; i can't recall them all... :/

**Comment by chrisbobbe** [comment]
The auto-generated Podfile (from `react-native init`) differs by version, but it sounds like the same "Integration with Existing Apps" doc is served no matter which react-native version you've selected on the website. I learned that from reading #33, and I made a note there.
